### PR TITLE
St 101

### PR DESCRIPTION
**Pull request by nazire-aydin**
Hi,
 
   Could you please code review for ST-101?

Thanks, and Regards

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,6 +23,11 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
+		<dependency>
+    <groupId>org.springframework.boot</groupId>
```
Comment (talataydin) [inline pom.xml:27]: You should be careful for code style in your code. You can select all code and then Ctrl + Shift + F click code to set code style. You should install to Google style in your ide.

File: src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import uk.co.nazire.model.ErrorDetails;
+
+@RestControllerAdvice
+public class GlobalExceptionHandling {
+	
+	@ExceptionHandler(DataNotFoundException.class)
+	@ResponseStatus(HttpStatus.NOT_FOUND)
+	public ErrorDetails dataNotFoundExceptionHandler(final DataNotFoundException ex,final WebRequest request) {
+		
+		ErrorDetails errorDetails = new ErrorDetails(ex.getMessage(),request.getDescription(false),
```
Comment (talataydin) [inline src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java:22]: You should to use space of the parameters.

File: src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+				                                     new Date(),HttpStatus.NOT_FOUND.value());
+				
+		return errorDetails;
+	}
+	@ExceptionHandler(DataAlreadyExistException.class)
+	@ResponseStatus(HttpStatus.CONFLICT)
+	public ErrorDetails dataAlreadyExistExceptionHandler(final DataAlreadyExistException ex,final WebRequest request) {
+		
+		ErrorDetails errorDetails = new ErrorDetails(ex.getMessage(),request.getContextPath(), new Date(),HttpStatus.CONFLICT.value());
```
Comment (talataydin) [inline src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java:31]: You should to use space of the parameters.

File: src/main/java/uk/co/nazire/model/Student.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Student {
+	
+	private static final AtomicLong counter = new AtomicLong();
+	
+	private Long id;
+	
+	//@NotEmpty
```
Comment (talataydin) [inline src/main/java/uk/co/nazire/model/Student.java:11]: Could you please check this validation?

File: src/main/java/uk/co/nazire/service/StudentService.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.List;
+
+import uk.co.nazire.model.Student;
+
+public interface StudentService {
+	
+	List<Student> getStudentList();
+	
+	Student getId(Long id);
+
```
Comment (talataydin) [inline src/main/java/uk/co/nazire/service/StudentService.java:12]: Be carefully for code style

File: src/main/java/uk/co/nazire/service/StudentServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+		return STUDENT_DATA;
+	}
+
+
+	@Override
+	public Student getId(Long id) {
+		return STUDENT_DATA.stream()
+				.filter(s -> s.getId().equals(id))
+				.findAny()
+				.orElseThrow(() -> new DataNotFoundException(id + "not found"));
```
Comment (talataydin) [inline src/main/java/uk/co/nazire/service/StudentServiceImpl.java:31]: You should to use space before not found.

File: src/test/java/uk/co/nazire/model/test/StudentTest.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+		assertNotEquals(student ,student1);
+	}
+	
+	@Test
+	public void getterTest() {
+		Student student = new Student("recep","solmaz",15,"PE");
+		
+		assertEquals("recep",student.getName());
+		assertEquals("solmaz",student.getSurName());
+
```
Comment (talataydin) [inline src/test/java/uk/co/nazire/model/test/StudentTest.java:38]: You can check the all values.

**Review by talataydin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (nazire-aydin) [reply pom.xml:27]: I installed. Thanks

File: src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java (hunk 2 above)
Comment (nazire-aydin) [reply src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java:22]: I have done. I should be carefully after that.

File: src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java (hunk 3 above)
Comment (nazire-aydin) [reply src/main/java/uk/co/nazire/controller/GlobalExceptionHandling.java:31]: I have done. I should be carefully after that.

File: src/main/java/uk/co/nazire/service/StudentService.java (hunk 5 above)
Comment (nazire-aydin) [reply src/main/java/uk/co/nazire/service/StudentService.java:12]: I have done. I should be carefully after that.

**Review by talataydin — APPROVED** [review]

**Comment by talataydin** [comment]
Hi,
 
   I approved your pull request.  We can merge to develop branch.

Thanks and Regards,

**Pull request merged**